integration-tests/tests/integration_test.rs

```diff
@@ -11766,6 +11766,49 @@ fn test_issue_1170() {
     run_test("", hdr, quote! {}, &["Arch"], &[]);
 }
 
+#[test]
+fn test_issue_1192() {
+    let hdr = indoc! {
+        "#include <vector>
+        #include <cstdint>
+        template <typename B>
+        struct A {
+            B a;
+        };
+        struct VecThingy {
+            A<uint32_t> contents[2];
+        };
+        struct MyStruct {
+            VecThingy vec;
+        };"
+    };
+    run_test_ex(
+        "",
+        hdr,
+        quote! {},
+        quote! {
+
+            extern_cpp_type!("VecThingy", crate::VecThingy)
+            pod!("VecThingy")
```

Comment:
Thinking about this a bit more: maybe this test should actually be failing, *unless* the target Rust type implements `Copy`? IIUC, that's basically what `pod!` expects?

Reply:
(as in, my example from the original issue should probably also fail because I forgot `#[derive(Clopy, Copy)]` which is the only guarantee that the type is, in fact, trivial)

Reply:
Thoughts?

Reply:
Sorry, no time for autocxx at the moment - I will reply next time I get back to the project. (However your mental model of how `pod!` works is wrong: I need to spend some time documenting exactly what are the expectations are for `extern_cpp_type!` users in the POD and non-POD case.)

Reply:
I improved the documentation here in #1221.

TL;DR: deriving things in Rust is not material to whether things can be POD or non-POD; it's all about the nature of the pre-existing C++ constructors/destructors, and by the time we get as far as `#[derive]` there's no way we can add extra C++ assertions.

Overall, this `extern_cpp_type` facility was intended primarily for one set of autocxx bindings to refer to a type generated by some other autocxx/cxx bindings, i.e. cross-mod links, rather than for folks such as your good self to generate whole new fresh shiny Rust types. I've clarified that too.